content/docs/0.16.x/manage/shipyard/index.md

```diff
@@ -1,290 +1,103 @@
-  properties: 
-    pass: automatic
-    warning: manual
-```
-
-**Note:** If the approval is the first task of a sequence, the approval service is not provided with the result from a previous task. In this case, it always falls back to the `manual` approval strategy.
-
-<details><summary>**Example:** Extended shipyard with a mandatory approval task in production</summary>
-
-<p>
-
-```yaml
-apiVersion: spec.keptn.sh/0.2.3
-kind: "Shipyard"
-metadata:
-  name: "shipyard-sockshop"
-spec:
-  stages:
-    - name: "production"
-      sequences:
-        - name: "delivery"
-          tasks:
+    apiVersion: spec.keptn.sh/0.2.3
+    kind: "Shipyard"
+    metadata:
+      name: "shipyard-sockshop"
+    spec:
+      stages:
+        - name: "dev"
+          sequences:
+          - name: "delivery"
+            tasks: 
             - name: "deployment"
-            - name: "approval"
-              properties:
-                pass: "manual"
-                warning: "manual"
             - name: "release"
-```
-
-</p>
-</details>
-
-#### Deployment
-
-Defines the deployment strategy (see [Continuous Delivery](../../continuous_delivery/)) used to deploy a new version of a service. Keptn supports deployment strategies of type:
-
-* `direct`: Deploys a new version of a service by replacing the old version of the service.
-* `blue_green_service`: Deploys a new version of a service next to the old one. After a successful validation of this new version, it replaces the old one and is marked as stable (i.e., it becomes the `primary`-version).
-* `user_managed` (experimental): Deploys a new version of a service by fetching the current helm-chart from the git repo, and just updating certain values.
-
-**Usage:**
-
-```
-- name: deployment
-  properties: 
-    deploymentstrategy: blue_green_service
-```
-
-```
-- name: deployment
-  properties: 
-    deploymentstrategy: direct
-```
-
-```
-- name: deployment
-  properties: 
-    deploymentstrategy: user_managed
-```
-
-#### Evaluation
-
-Defines the quality evaluation that is executed to verify the quality of a deployment based on its SLOs/SLIs.
-
-**Usage:**
-```
-- name: evaluation
-```
-
-#### Get-Action
-
-The get-action task is used to extract the desired remediation action from a remediation.yaml within a [remediation workflow](../../automated_operations/remediation).
-
-**Usage:**
-
-```
-- name: get-action
-```
-
-#### Release
 
-Defines the releasing task that is executed after a successful deployment happens. This means that production traffic is shifted towards the new deployment in this task.
-
-**Usage:**
-```
-- name: release
-```
-
-#### Action
-
-Defines the execution of a remediation action retrieved by `get-action`.
-
-**Usage:**
-```
-- name: action
-```
-
-#### Rollback
-
-Defines the rollback task that is executed when a rollback is triggered.
-
-**Usage:**
-```
-- name: rollback
-```
-
-#### Test
-
-Defines the test strategy used to validate a deployment. Keptn supports tests of type:
```

Comment:
It is only the `jmeter-service` that supports the `functional` and `performance` keywords. Indeed task properties are:
a) Entirely optional
b) Send for, processed by and only relevant for the tool processing the task event.

In other words, if you asked k6 to respond to a `test` event with a property set to `performance` it would be entirely ignored.

Perhaps move "task properties" to it's own page and entirely remove them from this page (ie. leave all shipyard files sans properties block)?

Reply:
I think I handled this in the reference page, for now, and we'll have to figure out the rest.

We need a sample shipyard file that has a test task that uses something other than jmeter.  I am assuming that it cannot be named "test" because that term is reserved for jmeter, right?  I pulled out the old "for historical reasons and backwards compatibility" trope to explain why these tasks are special ;-)  And then we need to create a page in the "Creating shipyard files" section about using test tools other than jmeter -- I'm not sure whether this all needs to be defined in the service for that tool or if I have some flexibility about how I incorporate it into my shipyard -- I suspect the latter.